lib/utils.pm

```diff
@@ -1120,4 +1121,54 @@ sub prepare_ssh_localhost_key_login {
     }
 }
 
+=head2 script_run_interactive
+For interactive command, input strings or keys according to the prompt message
+in the run time. Pass arrayref which contains the prompt message to be matched
+(regex) and the answers
+=cut
+sub script_run_interactive {
+    my ($cmd, $scan, $timeout) = @_;
+    my $output;
+    my @words;
+    $timeout //= 180;
+
+    if ($cmd) {
+        script_run("script -c \'", 0);
+        script_run($cmd,           0);
+
+        # Write to /dev/null since we want not to leave file there
+        script_run("\' /dev/null |& tee /dev/$serialdev", 0);
+    }
+
+    for my $k (@$scan) {
+        push(@words, $k->{prompt});
+    }
+
+    my $endmark = "Script done.*\/dev\/null";
+    push(@words, $endmark);
+
+  LOOP: {
+        do {
+            $output = wait_serial(\@words, $timeout) || die "No message matched!";
+
+            last LOOP if ($output =~ /$endmark/m);
+            for my $i (@$scan) {
+                if ($output =~ $i->{prompt}) {
+                    if ($i->{string}) {
+                        type_string $i->{string};
+                        last;
+                    }
+                    elsif ($i->{key}) {
+                        send_key $i->{key};
+                        last;
+                    }
+                    else {
+                        die "$i->{prompt} - No flags specified";
+                    }
```

Comment:
Could you try to reduce the complexity/indendation within this loop?

Reply:
For example you could write
```perl
next if ($output !~ $i->{prompt});
``` 
to reduce nested level by one.

Reply:
@okurz @rwx788  Thank you for the great suggestion, I learned a lot from you. Changes have been made. Please check it.

Reply:
Sorry if I could not come up with a more helpful, specific suggestion. Thanks to @rwx788 for the hint.